src/commands/analyze/utils.ts

```diff
@@ -162,32 +162,57 @@ export async function addServicesToConfig(configPath: string, services: YAMLServ
 }
 
 /**
- * Injects the backend function URLs into the frontend environment variables.
+ * Injects the backend function URLs like express, flask (backend functions), nestjs and nitro
+ * into the frontend environment variables like react, vue, angular, nextjs and nuxt.
+ *
  * @param frontendPrefix The prefix to use for the frontend environment variables.
  * @param configPath The path to the config file.
  */
-export async function injectBackendApiUrlsInConfig(configPath: string, frontendPrefix: string) {
+export async function injectBackendUrlsInConfig(configPath: string) {
     const configIOController = new YamlConfigurationIOController(configPath);
 
     // Load config with minimal changes
     const config = await configIOController.read(/* fillDefaults= */ false);
 
-    // Validate backend and frontend existence
-    let functions: string[] = [];
-    const ssrFrameworks: string[] = [];
     const backend = config.backend as YAMLBackend;
-    const frontend = config.frontend as YamlFrontend;
-    if (!frontend) return;
-
-    if (backend?.functions) functions = backend.functions.map((fn) => fn.name);
-    if (config.nextjs) ssrFrameworks.push("nextjs");
-    if (config.nestjs) ssrFrameworks.push("nestjs");
-    if (config.nuxt) ssrFrameworks.push("nuxt");
-    if (config.nitro) ssrFrameworks.push("nitro");
+    const functions: string[] = backend?.functions?.map((fn) => fn.name) || [];
+    const ssrFunctions: string[] = [];
+    if (config.nestjs) ssrFunctions.push("nestjs");
+    if (config.nitro) ssrFunctions.push("nitro");
```

Comment:
I think this approach could be easier to maintain, and the ssrs differs only by framework name.

'''

const ssrFrameworks = ["nestjs", "nitro"];
ssrFrameworks.forEach((framework) => {
    if (config[framework]) ssrFunctions.push(framework);
});

'''